filebeat/harvester/log.go

```diff
@@ -255,3 +226,21 @@ func (h *Harvester) updateOffset(increment int64) {
 func (h *Harvester) UpdateState() {
 	h.sendEvent(h.createEvent())
 }
+
+func (h *Harvester) close() {
+	// On completion, push offset so we can continue where we left off if we relaunch on the same file
+	if h.Stat != nil {
+		h.Stat.Offset <- h.getOffset()
```

Comment:
Long term h.Stat will potentially completely disappear, so will this part of the code. The offset should be sent with every even so no special offset reporting will be needed.